From f2e47bc333654e96dfd2f1e05435ba80c28c7ec9 Mon Sep 17 00:00:00 2001
From: Sridharan S N <sridsn@qti.qualcomm.com>
Date: Mon, 24 Aug 2026 16:01:20 +0530
Subject: [PATCH] soc: qcom: socinfo: Add sysFS for QCLIB info

With this update:

IMAGE_INDEX_SBL (0) will be used for SPL version information.
IMAGE_INDEX_QCLIB (6) will be used for QCLIB version information.

Change-Id: I3dbb5f95d301bd03d5bfda64e05fd7ebc7461f03
Signed-off-by: Sridharan S N <sridsn@qti.qualcomm.com>
(cherry picked from commit 1053a60da1854b08bafc59bc2e296af0d84e28db)
---
 drivers/soc/qcom/socinfo.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/soc/qcom/socinfo.c b/drivers/soc/qcom/socinfo.c
index dc2465841a47..5aca0ba3ede1 100644
--- a/drivers/soc/qcom/socinfo.c
+++ b/drivers/soc/qcom/socinfo.c
@@ -42,6 +42,7 @@
 #define SMEM_IMAGE_TABLE_BOOT_INDEX     0
 #define SMEM_IMAGE_TABLE_TZ_INDEX       1
 #define SMEM_IMAGE_TABLE_RPM_INDEX      3
+#define SMEM_IMAGE_TABLE_QCLIB_INDEX	6
 #define SMEM_IMAGE_TABLE_APPS_BL_INDEX	9
 #define SMEM_IMAGE_TABLE_APPS_INDEX     10
 #define SMEM_IMAGE_TABLE_MPSS_INDEX     11
@@ -61,6 +62,7 @@ static const char *const socinfo_image_names[] = {
 	[SMEM_IMAGE_TABLE_BOOT_INDEX] = "boot",
 	[SMEM_IMAGE_TABLE_CNSS_INDEX] = "cnss",
 	[SMEM_IMAGE_TABLE_MPSS_INDEX] = "mpss",
+	[SMEM_IMAGE_TABLE_QCLIB_INDEX] = "qclib",
 	[SMEM_IMAGE_TABLE_RPM_INDEX] = "rpm",
 	[SMEM_IMAGE_TABLE_TZ_INDEX] = "tz",
 	[SMEM_IMAGE_TABLE_VIDEO_INDEX] = "video",
-- 
2.34.1

